File: day-02/part1.py
from util.run import run
from util.direction.direction import Direction, InvalidDirection


def solution(moves: list[tuple[Direction, int]]) -> int:
    horizontal = 0
    depth = 0

    for (dir, scalar) in moves:
        if dir == Direction.FORWARD:
            horizontal += scalar
        elif dir == Direction.DOWN:
            depth += scalar
        elif dir == Direction.UP:
            depth -= scalar
        else:
            raise InvalidDirection()

    return horizontal * depth


if __name__ == "__main__":
    run(solution)

File: day-02/part1.go
package main

import (
	"github.com/dawidl022/advent-of-code-2021/day-02/util"
	"github.com/dawidl022/advent-of-code-2021/day-02/util/direction"
)

func solution1(moves []direction.Move) int {
	depth := 0
	horizontal := 0

	for _, move := range moves {
		switch move.Dir {
		case direction.FORWARD:
			horizontal += move.Scalar
		case direction.DOWN:
			depth += move.Scalar
		case direction.UP:
			depth -= move.Scalar
		}
	}
	return horizontal * depth
}

func main() {
	util.Run(solution1)
}
